datafusion/functions/src/core/named_struct.rs

```diff
@@ -57,6 +57,15 @@ fn named_struct_expr(args: &[ColumnarValue]) -> Result<ColumnarValue> {
         .into_iter()
         .unzip();
 
+    // Check to enforce the uniqueness of struct field name
+    let unique_field_names_set = names.iter().collect::<HashSet<_>>();
+
+    if unique_field_names_set.len() != names.len() {
+        return exec_err!("named_struct requires unique field names");
```

Comment:
It might be nice if you could structure the code so it gave one of the duplicated names. Maybe something like

```rust
let mut unique_field_names_set = HashSet::new();
for name in names {
  if unique_field_names_set.contains(name) {
        return exec_err!("named_struct requires unique field names. {name} repeated");
  }
  unique_field_names_set.insert(name)
}
```

Reply:
That's a great idea @alamb . Thank You. I will make the changes